crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -683,7 +683,16 @@ where
 
 fn pattern() -> impl NoirParser<Pattern> {
     recursive(|pattern| {
-        let ident_pattern = ident().map(Pattern::Identifier);
+        let ident_pattern = ident().map(Pattern::Identifier).map_err(|mut error| {
+            if matches!(error.found(), Token::IntType(..)) {
+                error = ParserError::with_reason(
+                    ParserErrorReason::ExpectedPattern(error.found().clone()),
+                    error.span(),
+                );
+            }
```

Comment:
I'm not the biggest fan of hard coding specific issues in the parser rather than trying to fix them more generally. I suppose the confusion was somewhat specific to `i2` being an integer type where it isn't in most languages though. Instead of changing the error entirely perhaps we can keep the check and add a secondary message conditionally along the lines of "i2 is a type and cannot be used as a variable name"?